### PR TITLE
Sweep too many inputs in multiple transactions (Sweeper)

### PR DESCRIPTION
**Pull request by lacksfish**
Please review, discuss and merge :)

### CODE REVIEW

File: src/js/modules/core/services/sweeper/sweeper.service.js [hunk 1]
```diff
@@ -102,30 +97,58 @@
             }
 
             return bitcoinDataClient.estimateFee().then(function (feePerKB) {
-                if (totalValue <= blocktrailSDK.DUST) {
-                    return false;
-                }
 
-                rawTransaction.addOutput(options.recipient, totalValue);
+                var txAfter = 600;
```
Comment (afk11) [inline src/js/modules/core/services/sweeper/sweeper.service.js]: Maybe add a comment describing where this number came from (n148+56 I think), it depends on the utxo type never changing, etc

File: src/js/modules/core/services/sweeper/sweeper.service.js [hunk 2]
```diff
@@ -102,30 +97,58 @@
...
-                rawTransaction.tx.outs[0].value -= fee;
+                var chunks = [];
+                var index = 0;
+                var length = inputs.length;
 
-                if (rawTransaction.tx.outs[0].value <= blocktrailSDK.DUST) {
-                    return false;
+                while (index < length) {
+                    chunks.push(inputs.slice(index, index += txAfter + 1));
+                    // index += txAfter;
```
Comment (afk11) [inline src/js/modules/core/services/sweeper/sweeper.service.js]: the assignment should probably be moved out of `inputs.slice`.. could it be rewritten as a for loop?

File: src/js/modules/core/services/sweeper/sweeper.service.js [hunk 3]
```diff
@@ -102,30 +97,58 @@
...
+                        totalValue += parseInt(input['value']);
+                        rawTransaction.addInput(input['txid'], input['vout']);
+                    });
+
+                    if (totalValue <= blocktrailSDK.DUST) {
+                        return false;
+                    }
+
+                    rawTransaction.addOutput(options.recipient, totalValue);
+                    console.log("FeePerKb: ", feePerKB);
```
Comment (afk11) [inline src/js/modules/core/services/sweeper/sweeper.service.js]: this is one of mine I think..

File: src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js [hunk 4]
```diff
@@ -90,15 +97,26 @@
...
+                    return !!result;
+                });
+
+                // Add to absolute value to display before sending
+                results.map(function (result) {
+                    $scope.displaySweepData.totalValue += result.totalValue;
+                    $scope.displaySweepData.totalFeePaid += result.feePaid;
+                });
+
+                if (results && results.length && somePositiveBalances) {
```
Comment (afk11) [inline src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js]: I notice this part, and the next part below are very similar.. could make the same function perhaps. 

sweeperService.wifSweep(WIFs, options).then($scope.displayResult)
sweeperService.bip44Sweep(seed, options).then($scope.displayResult)

File: src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js [hunk 5]
```diff
@@ -127,15 +145,26 @@
...
```
Comment (afk11) [inline src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js]: duplicated code? see note above

File: src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js [hunk 6]
```diff
@@ -158,23 +187,47 @@
...
+                        }
+                    });
+
+                    if (!failures) {
+                        $timeout(function () {
+                            // TODO: Display on PUBLISHED STATE MULTIPLE
+                            $scope.form.step = $scope.STEPS.PUBLISH;
+                        });
+                        trackingService.trackEvent(trackingService.EVENTS.SWEEP.SWEEP_SUCCESS);
+                    } else {
```
Comment (afk11) [inline src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js]: This says if there are no failures show success, but we may have already broadcast 1 of the two transaction by then. It's due to an unlikely amount, but if we don't deal with this in an if, the else will catch it, which would say the it failed?

File: src/js/modules/core/services/sweeper/sweeper.service.js (hunk 2 above)
Comment (lacksfish) [reply src/js/modules/core/services/sweeper/sweeper.service.js]: Yes, that's possible. Done.

File: src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js (hunk 4 above)
Comment (lacksfish) [reply src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js]: Done, `then` now calls `processSweepResults()`

File: src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js (hunk 5 above)
Comment (lacksfish) [reply src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js]: Done, `then` now calls `processSweepResults()`

File: src/js/modules/core/services/sweeper/sweeper.service.js (hunk 3 above)
Comment (lacksfish) [reply src/js/modules/core/services/sweeper/sweeper.service.js]: Yup, removed. :)

File: src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js (hunk 6 above)
Comment (lacksfish) [reply src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js]: "TX_CANT_BE_PUSHED_MULTIPLE": "{{ amount }} of the transactions can not be pushed. An error occurred."

Would be the new translation for one or more tx's failing.

**Pull request merged**